### DISCUSSION TITLE
Allow usage only on home-manager

### DISCUSSION THREAD

**Discussion by dwarfmaster** [nix-community/stylix#21] [Ideas]
Hi,

I would like if it was possible to use stylix both with and without NixOS. The way I am thinking is to duplicate all colors/fonts options in home-manager, and change the NixOS module to set the options to the same value as the corresponding options in NixOS. Furthermore, all targets should be split between a pure home-manager module, and a NixOS one that simply enable the home-manager one (if the module does set something in home-manager of course).

Why do I want that ? Because in my configuration my home-manager modules do not have access to the nixos config, so if I want to theme an application using stylix palette I do not have access to it. Furthermore, if I want to enable a specific stylix configuration, for example for alacritty, I must do so from a nixos module, which breaks the separation I arrived at in my config.

This would also enable people using home-manager without nixos to use stylix, and even more this would enable multiple users on a single machine to use different themes.

I would like to have your opinion about this. If you believe this is a good idea I am willing to refactor the current code and open a PR.

**Comment by danth** [comment]
Allowing multiple users with different themes sounds like a great idea, and something I thought about in the past.

It might be best to have two separate entry points:

- A NixOS module importing all the system wide targets
- A Home Manager module importing all the user specific targets

Then in addition to using Home Manager without NixOS, it will be possible to use NixOS without Home Manager.

The NixOS module would optionally copy the system theme into Home Manager, if they are configured together. We can check whether this is the case using `options?home-manager`, which will only be true if Home Manager is imported into the system configuration.

Some of the modules currently rely on system wide options, where it should be possible to use Home Manager instead. It would be beneficial to convert these.

Please do go ahead and open a PR! I can contribute some of those conversions once you have the initial refactoring in place.

**Comment by dwarfmaster** [comment]
PR opened: #23

**Reply by trueNAHO** [reply]
This has been merged.
